extensions/groovy-actions/rest/src/main/java/org/apache/unomi/groovy/actions/rest/GroovyActionsEndPoint.java

```diff
@@ -0,0 +1,103 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.unomi.groovy.actions.rest;
+
+import org.apache.cxf.jaxrs.ext.multipart.Attachment;
+import org.apache.cxf.jaxrs.ext.multipart.Multipart;
+import org.apache.cxf.rs.security.cors.CrossOriginResourceSharing;
+import org.apache.unomi.groovy.actions.services.GroovyActionsService;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jws.WebMethod;
+import javax.jws.WebService;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+@WebService
+@Produces(MediaType.APPLICATION_JSON + ";charset=UTF-8")
+@CrossOriginResourceSharing(allowAllOrigins = true, allowCredentials = true)
+@Path("/groovyActions")
+@Component(service = GroovyActionsEndPoint.class, property = "osgi.jaxrs.resource=true")
+public class GroovyActionsEndPoint {
+
+    private static final Logger logger = LoggerFactory.getLogger(GroovyActionsEndPoint.class.getName());
+
+    @Reference
+    private GroovyActionsService groovyActionsService;
+
+    public GroovyActionsEndPoint() {
+        logger.info("Initializing groovy actions service endpoint...");
+    }
+
+    @WebMethod(exclude = true)
+    public void setGroovyActionsService(GroovyActionsService groovyActionsService) {
+        this.groovyActionsService = groovyActionsService;
+    }
+
+    /**
+     * Save a groovy action file and create an actionType entry to allow to call this action
+     *
+     * @param file the file to save
+     * @return
+     */
+    @POST
+    @Path("/")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response save(@Multipart(value = "file") Attachment file) {
+        try {
+            String filename = file.getContentDisposition().getParameter("filename");
+
+            java.nio.file.Path path = Paths.get("/tmp/" + filename);
```

Comment:
I don't think this way of retrieving the temporary directory is correct. Instead we should use something like this: https://attacomsian.com/blog/java-create-temporary-directory#:~:text=In%20Java%207%20and%20higher%2C%20you%20can%20use,and%20print%20its%20absolute%20path%20on%20the%20console. and create a "unomi-groovy-actions" directory to make sure we don't mess up the main temp directory.